File: typescript/packages/plugins/spl-token/src/tokens.ts
export type SolanaNetwork = "devnet" | "mainnet";

export type Token = {
    decimals: number;
    symbol: string;
    name: string;
    mintAddresses: Record<SolanaNetwork, string | null>;
};

export type NetworkSpecificToken = Omit<Token, "mintAddresses"> & {
    network: SolanaNetwork;
    mintAddress: string;
};

export const USDC: Token = {
    decimals: 6,
    symbol: "USDC",
    name: "USDC",
    mintAddresses: {
        devnet: "4zMMC9srt5Ri5X14GAgXhaHii3GnPAEERYPJgZJDncDU",
        mainnet: "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v",
    },
};

export const GOAT: Token = {
    decimals: 6,
    symbol: "GOAT",
    name: "GOAT",
    mintAddresses: {
        mainnet: "CzLSujWBLFsSjncfkh59rUFqvafWcY5tzedWJSuypump",
        devnet: null,
    },
};

export const PENGU: Token = {
    decimals: 6,
    symbol: "PENGU",
    name: "Pengu",
    mintAddresses: {
        mainnet: "2zMMhcVQEXDtdE6vsFS7S7D5oUodfJHE8vd1gnBouauv",
        devnet: null,
    },
};

export const SOL: Token = {
    decimals: 9,
    symbol: "SOL",
    name: "Wrapped SOL",
    mintAddresses: {
        mainnet: "So11111111111111111111111111111111111111112",
        devnet: "So11111111111111111111111111111111111111112",
    },
};

export const SPL_TOKENS: Token[] = [USDC, GOAT, PENGU, SOL];

File: python/src/wallets/solana/goat_wallets/solana/tokens.py
from typing import Dict, List, Literal

from goat.types.token import Token as CoreToken

class Token(CoreToken):
    mintAddress: str

SolanaNetwork = Literal["mainnet", "devnet", "testnet"]

USDC: Token = {
    "name": "USD Coin",
    "symbol": "USDC",
    "decimals": 6,
    "mintAddress": "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v",
}

USDT: Token = {
    "name": "Tether USD",
    "symbol": "USDT",
    "decimals": 6,
    "mintAddress": "Es9vMFrzaCERmJfrF4H2FYD4KCoNkY11McCe8BenwNYB",
}

BONK: Token = {
    "name": "Bonk",
    "symbol": "BONK",
    "decimals": 5,
    "mintAddress": "DezXAZ8z7PnrnRJjz3wXBoRgixCa6xjnB7YaB1pPB263",
}

USDC_DEVNET: Token = {
    "name": "USD Coin (Devnet)",
    "symbol": "USDC",
    "decimals": 6,
    "mintAddress": "4zMMC9srt5Ri5X14GAgXhaHii3GnPAEERYPJgZJDncDU",
}

SPL_TOKENS: Dict[SolanaNetwork, List[Token]] = {
    "mainnet": [USDC, USDT, BONK],
    "devnet": [USDC_DEVNET],
    "testnet": [],
}
